internal/brokers/broker.go

```diff
@@ -104,15 +104,27 @@ func (b Broker) SelectAuthenticationMode(ctx context.Context, sessionID, authent
 
 // IsAuthorized calls the broker corresponding method, stripping broker ID prefix from sessionID.
 func (b Broker) IsAuthorized(ctx context.Context, sessionID, authenticationData string) (access string, userInfo string, err error) {
-	sessionID = strings.TrimPrefix(sessionID, fmt.Sprintf("%s-", b.ID))
-
-	access, userInfo, err = b.brokerer.IsAuthorized(ctx, sessionID, authenticationData)
-	if err != nil {
-		return "", "", err
+	sessionID = b.parseSessionID(sessionID)
+
+	// monitor ctx in goroutine to call cancel
+	done := make(chan bool, 1)
```

Comment:
and without sending anything, you can remove the need to have a buffer channel. (and use directly a chan(struct{}))